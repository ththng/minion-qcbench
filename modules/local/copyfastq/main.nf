process COPYFASTQ {
    tag "$meta.id"
    label 'process_single'

    input:
    tuple val (meta), path (fastq)

    output:
    tuple val (meta), path ("*.fastq.gz"), emit: fastq

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"


    """
    cp $args $fastq ${prefix}.fastq.gz
    """
    
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    echo stub | gzip -c > ${prefix}.fastq.gz
    """
}
