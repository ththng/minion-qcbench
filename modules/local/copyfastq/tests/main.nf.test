nextflow_process {

    name "Test Process COPYFASTQ"
    script "../main.nf"
    process "COPYFASTQ"

    test("Should copy the file correctly") {

        when {
            params {
                outdir = "results"
            }
            process {
                """
                input[0] = [ [id: 'testsample' ], file("${projectDir}/tests/data/test.fastq.gz") ]
                """
            }
        }

        then {
            assert process.success

            // Check if the output file has the correct name
            assert file(process.out.fastq[0][1]).name == "testsample_noqc.fastq.gz"

            // Check if the input and output file have the same md5sum
            assert path(process.out.fastq[0][1]).md5 == path("${projectDir}/tests/data/test.fastq.gz").md5

            assert snapshot(process.out).match()
        }

    }

}
