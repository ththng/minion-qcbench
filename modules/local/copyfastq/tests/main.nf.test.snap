{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "testsample"
                        },
                        "testsample_noqc.fastq.gz:md5,18199a800dd991e815b9f6c84935f1ca"
                    ]
                ],
                "fastq": [
                    [
                        {
                            "id": "testsample"
                        },
                        "testsample_noqc.fastq.gz:md5,18199a800dd991e815b9f6c84935f1ca"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-20T16:53:51.65008"
    }
}